### DISCUSSION TITLE
Remove title bars from all windows

### DISCUSSION THREAD

**Discussion by cgeoga** [YaLTeR/niri#404] [Q&A] [answered]
Hello,

Very long-time i3 user here. In i3/X, I am able to effectively disable window titles in i3 with `default_border pixel 1` in my config. I'd love to somehow reproduce this behavior of globally turning off all title bars for every window, even if it is an ugly hack.

From some googling, it seems like in Wayland title bars for windows are controlled by each individual application and not Wayland itself. Sway seems to get around this by setting the font of the title to zero or near-zero ([ref](https://github.com/swaywm/sway/issues/7350)), but that feature doesn't seem to be stable.

I've looked at all the issues, discussions, and read the config file for Niri, but I haven't really seen any information about a way to achieve this. Can anybody advise? Thanks in advance for your time and thoughts and apologies if I have missed any earlier discussions about this.

**Comment by lpnh** [comment, marked as answer]
Hi! Try to uncomment the `prefer-no-csd` line and see if it works.

https://github.com/YaLTeR/niri/blob/ae7fb4c4f405aa0ff49930040d414581a812d938/resources/default-config.kdl#L198

**Comment by cgeoga** [comment]
So embarrassing---I could have sworn I tried that and concluded it didn't work, but it definitely does and looks great. My apologies for the post and thanks for creating such a fun and pleasant WM. After experimenting for a few hours I'm definitely making the switch!

**Reply by lpnh** [reply]
No worries! I share your appreciation for YaLTeR's wonderful work. Niri is incredibly polished considering how new the project is.

**Reply by YaLTeR** [reply]
Glad it worked! 

> I could have sworn I tried that and concluded it didn't work

Likely because it requires restarting the clients after you change it, which is mainly due to an issue where many SDL2 versions will freeze if implemented differently.
